src/Model/Volunteer.hs

```diff
@@ -0,0 +1,19 @@
+module Model.Volunteer where
+
+import Import
+
+import Control.Monad.Trans.Writer.Strict (tell)
+
+fetchAllInterestsDB :: DB [Entity Interest]
+fetchAllInterestsDB = selectList [] []
+{-# INLINE fetchAllInterestsDB #-}
+
+-- | Record a volunteer application.
+insertVolunteerApplicationDB :: ProjectId -> VolunteerApplication -> [InterestId] -> SDB ()
+insertVolunteerApplicationDB project_id application interest_ids = do
+    application_id <- lift (insert application)
+    forM_ interest_ids $ \interest_id ->
+        lift (insert (VolunteerInterest application_id interest_id))
```

Comment:
not sure how SDB works, but these inserts (along with the application insert) should be in a transaction if they aren't